app/controllers/sequences_controller.rb

```diff
@@ -179,14 +185,16 @@ def set_sequence
   end
 
   def find_or_create_sequence_run
-    if sequence_run_id = params['sequence_run']
-      @sequence_run = SequenceRun.find(sequence_run_id)
+    if sequence_run_key = params['sequence_run_key']
+      @sequence_run = SequenceRun.find_by_key(sequence_run_key)
```

Comment:
if an invalid sequence_run_key is used a 404 should be returned, the code should not continue and create a new sequence run